apixu-weatherbot1.0.tcl

```diff
@@ -0,0 +1,498 @@
+    http::register https 443 [list tls::socket -tls1 1 -ssl2 0 -ssl3 0]
+
+    setudef flag weather-apixu
+    # Default units for weather output. 0 for metric, 1 for imperial, 2 for both.
+    variable units "1"
+    variable base_url "https://api.apixu.com/v1"
+    # Set your apikey here
+    variable apikey "<insert-your-key-here>"
+    # Set to 1 (0 by default) if you want set weather responses
+    # to use PM/notify. Set to 0 if you want it all in the channel.
+    variable private 0
+
+    # Public channel command bindings
+    bind pub - .wz weather::current
+    bind pub - .wzf weather::forecast
+    bind pub - .set weather::location
+
+    # Private message command bindings
+    bind msg - .wz weather::pm_current
+    bind msg - .wzf weather::pm_forecast
+    bind msg - .set weather::pm_location
+
+    ##########################
+    # Public command functions
+    ##########################
+    proc current {nick uhost hand chan text} {
+        set text [string trim $text]
+
+        if {$text eq "--help" || $text eq "-h"} {
+            _get_help $nick
+            return
+        }
+
+        # Checks to see if a user has a location set
+        set userinfo [_get_userinfo $nick $hand]
+        if {$text eq "" && [dict exists $userinfo error]} {
+            if {$::weather::private} {
+                puthelp "NOTICE $nick :Did you want the weather for a specific location?\
+                         Or please PM me with the \".set\" command to set a default location."
+                return
+            }
+
+            puthelp "PRIVMSG $chan :No location set or use \002'.wz <location>'\002"
+            return
+        } elseif {[dict exists $userinfo error]} {
+            set userinfo [dict create location $text units $::weather::units]
+        } elseif {$text ne ""} {
+            set userinfo [dict create location $text units [dict get $userinfo units]]
+        }
+
+        set location [dict get $userinfo location]
+        if {[catch {set json_data [_get_json $location "current"]} errormsg]} {
+            puthelp "PRIVMSG $chan :$errormsg"
+            return
+        }
+
+        if {[catch {set current_weather [_json_parse $json_data $userinfo "current"]} errormsg]} {
+            puthelp "PRIVMSG $chan :$errormsg"
+            return
+        }
+
+        puthelp "PRIVMSG $chan :$current_weather"
+    }
+
+    proc forecast {nick uhost hand chan text} {
+        set text [string trim $text]
+
+        if {$text eq "--help" || $text eq "-h"} {
+            _get_help $nick
+            return
+        }
+
+
+        # Checks to see if a user has a location set
+        set userinfo [_get_userinfo $nick $hand]
+        if {$text eq "" && [dict exists $userinfo error]} {
+            if {$::weather::private} {
+                puthelp "NOTICE $nick :Did you want the weather for a specific location?\
+                         Or please PM me with the \".set\" command to set a default location."
```

Comment:
why not give the user a command they can just copy paste? like /msg botname (if you have it) .set ... rest of the syntax for a zip code or obvious airport code

Reply:
You're right about that, I have it in the other messages. Why not that one....idk? good catch